include/LinearAlgebra/sparseLUSolver.h

```diff
@@ -192,6 +192,21 @@ void SparseLUSolver<T>::factorizeWithHashing(const SparseMatrixCSR<T>& A)
             else
                 U_map[i][j] = val;
         }
+
+        const T tolerance_abs = 1e-12; // minimum allowed diagonal
+        const T tolerance_rel = 1e-8; // relative to the max in the row
```

Comment:
I am surprised that these values are hard coded. Especially given that this class is templated by `T`. If you try to do a low precision solve, these values are likely to be problematic

Reply:
My preference would be for these to be default parameters of the constructor. That way they can be modified to match the precision if necessary.